src/sidebar/services/test/root-thread-test.js

```diff
@@ -255,11 +252,7 @@ describe('rootThread', function() {
 
     it('filter matches only notes when Notes tab is selected', function() {
       fakeBuildThread.reset();
-
-      fakeStore.state = Object.assign({}, fakeStore.state, {
-        selectedTab: uiConstants.TAB_NOTES,
-      });
-
+      fakeStore.state.selection.selectedTab = uiConstants.TAB_NOTES;
```

Comment:
I'm a little worried here that seeing code like this in these tests, where it happens to work fine, might make developers less familiar with Redux think it is OK to write code like this in the actual application, where it would break important invariants.

We use "seamless-immutable" to guard against accidental mutation of state in some tests. It might be a good idea to start making use of it in the actual application in development mode as well.